packages/fx-core/src/common/local/localEnvManager.ts

```diff
@@ -40,6 +51,14 @@ class LocalEnvManager {
     });
   }
 
+  // Load local settings without encryption
+  private async getRawLocalSettings(projectPath: string): Promise<Json | undefined> {
```

Comment:
Some methods may be called at the same moment, such as `getProgrammingLanguage`, `getProgrammingLanguage`. Is there a way to reduce the times of reading file?

Reply:
Hard to do so in `LocalEnvManager` since it's stateless.
Adjusted parameters to let caller to decide whether to read file once or multiple times.
Moved some user tasks logic to v2 api to apply v2 context.